### DISCUSSION TITLE
Fail streaming rpc from client

### DISCUSSION THREAD

**Discussion by MixusMinimax** [hyperium/tonic#1302] [Q&A]
In grpc, calls can send Status errors in both directions.
I have this code:

```rs
mut client: SomethingClient<Channel> = ...;

let outbound = async_stream::stream! {
    yield SomethingRequest { a: 1 };
    yield SomethingRequest { a: 2 };
    // if something happens, I want to fail with a status now, something like
    // return Err(Status::internal("asd"));
};

let response = client.something(Request::new(outbound)).await?;
```

It would be nice if I could use `try_stream`, but that does not work, since `Request` wants a stream that always returns requests directly.

I have sadly not found any information on it, since the examples don't ever do something like this.

Once the stream closes, the server could still be doing something, and should be notified by sending an error

**Comment by LucioFranco** [comment]
gRPC does not support fail-liable client streams so it doesn't accept Status on purpose. You would need to encode what ever failure mode into your protobuf message.

**Comment by gitmalong** [comment]
> gRPC does not support fail-liable client streams so it doesn't accept Status on purpose. You would need to encode what ever failure mode into your protobuf message.

Is the same valid for cancellation from server side? https://github.com/hyperium/tonic/discussions/1590
